pkg/tests/ossm/smoke_test.go

```diff
@@ -58,50 +55,56 @@ func TestSmoke(t *testing.T) {
 
 		oc.RecreateNamespace(t, meshNamespace)
 
-		t.LogStep("Install bookinfo pods and sleep pod")
-		app.InstallAndWaitReady(t, app.Bookinfo(ns), app.SleepNoSidecar(ns))
-
-		t.NewSubTest(fmt.Sprintf("install bookinfo with smcp %s", fromVersion)).Run(func(t TestHelper) {
+		t.NewSubTest(fmt.Sprintf("upgrade %s to %s", fromVersion, toVersion)).Run(func(t TestHelper) {
+			t.Logf("This test checks whether SMCP becomes ready after it's upgraded from %s to %s and bookinfo is still working after the upgrade and also test a clean installation of the target SMCP", fromVersion, toVersion)
+			t.Cleanup(func() {
+				app.Uninstall(t, app.Bookinfo(ns), app.SleepNoSidecar(ns))
+				oc.RecreateNamespace(t, meshNamespace)
+			})
 
-			t.LogStepf("Create SMCP %s and verify it becomes ready", fromVersion)
+			t.LogStepf("Install SMCP %s and verify it becomes ready", fromVersion)
 			assertSMCPDeploysAndIsReady(t, fromVersion)
-			if env.GetSMCPVersion().GreaterThan(version.SMCP_2_2) {
-				assertRoutesExist(t)
-			}
 
-			t.LogStep("Restart all pods to verify proxy is injected in all pods of Bookinfo")
-			oc.RestartAllPods(t, ns)
-			retry.UntilSuccess(t, func(t test.TestHelper) {
-				assertSidecarInjectedInAllBookinfoPods(t, ns)
-			})
+			t.LogStep("Install bookinfo pods and sleep pod")
+			app.InstallAndWaitReady(t, app.Bookinfo(ns), app.SleepNoSidecar(ns))
 
-			t.LogStep("Check if bookinfo productpage is running through the Proxy")
+			t.LogStep("Check if bookinfo traffic flows through the Proxy")
 			assertTrafficFlowsThroughProxy(t, ns)
 
-			t.LogStep("verify proxy startup time. Expected to be less than 10 seconds")
-			t.Log("Jira related: https://issues.redhat.com/browse/OSSM-3586")
-			t.Log("From proxy json , verify the time between status.containerStatuses.state.running.startedAt and status.conditions[type=Ready].lastTransitionTime")
-			t.Log("The proxy startup time should be less than 10 seconds for ratings pod")
-			assertProxiesReadyInLessThan10Seconds(t, ns)
-		})
-
-		t.NewSubTest(fmt.Sprintf("upgrade %s to %s", fromVersion, toVersion)).Run(func(t TestHelper) {
-			t.Logf("This test checks whether SMCP becomes ready after it's upgraded from %s to %s and bookinfo is still working after the upgrade", fromVersion, toVersion)
-
 			t.LogStepf("Upgrade SMCP from %s to %s", fromVersion, toVersion)
 			assertSMCPDeploysAndIsReady(t, toVersion)
 			assertRoutesExist(t)
```

Comment:
Why isn't this part of `checkSMCP`?

Reply:
Actually, I'd add `assertTrafficFlowsThroughProxy` and `oc.RestartAllPodsAndWaitReady(t, "bookinfo")` to checkSMCP.

Reply:
Changed